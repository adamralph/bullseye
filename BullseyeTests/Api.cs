namespace BullseyeTests
{
    using System;
    using System.Linq;
    using Bullseye;
    using BullseyeTests.Infra;
    using PublicApiGenerator;
    using Xunit;

    public class Api
    {
        //[Fact]
        //public void GenerateApi() =>
        //    System.IO.File.WriteAllText(@"c:\src\api.txt", ApiGenerator
        //            .GeneratePublicApi(
        //                typeof(Targets).Assembly,
        //                typeof(Targets).Assembly.GetExportedTypes().Where(type => !type.Namespace.Contains("Internal")).ToArray())
        //            .Replace(Environment.NewLine, "\r\n"));

        [Fact]
        public void IsUnchanged() =>
            AssertFile.Contains(
                "api.txt",
                ApiGenerator
                    .GeneratePublicApi(
                        typeof(Targets).Assembly,
                        typeof(Targets).Assembly.GetExportedTypes().Where(type => !type.Namespace.Contains("Internal")).ToArray())
                    .Replace(Environment.NewLine, "\r\n"));
    }
}
