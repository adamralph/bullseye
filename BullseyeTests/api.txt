[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace Bullseye
{
    public interface IBuildContext
    {
        System.Collections.Generic.Dictionary<string, object> Items { get; }
    }
    public class static Targets
    {
        public static string[] DependsOn(params string[] dependencies) { }
        public static TInput[] ForEach<TInput>(params TInput[] inputs) { }
        public static void RunTargetsAndExit(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static System.Threading.Tasks.Task RunTargetsAndExitAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null, Bullseye.IBuildContext context = null) { }
        public static void RunTargetsWithoutExiting(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null, Bullseye.IBuildContext context = null) { }
        public static System.Threading.Tasks.Task RunTargetsWithoutExitingAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null, Bullseye.IBuildContext context = null) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, Bullseye.IBuildContext context) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<Bullseye.IBuildContext, System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<Bullseye.IBuildContext, System.Threading.Tasks.Task> action, Bullseye.IBuildContext context) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, Bullseye.IBuildContext, System.Threading.Tasks.Task> action, Bullseye.IBuildContext context) { }
        public static void Target(string name, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Func<Bullseye.IBuildContext, System.Threading.Tasks.Task> action, Bullseye.IBuildContext context) { }
        public static void Target(string name, System.Action action) { }
        public static void Target(string name, System.Action<Bullseye.IBuildContext> action, Bullseye.IBuildContext context = null) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, Bullseye.IBuildContext, System.Threading.Tasks.Task> action, Bullseye.IBuildContext context) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput, Bullseye.IBuildContext> action, Bullseye.IBuildContext context) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        public static void Target<TDummy>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action<Bullseye.IBuildContext> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action<Bullseye.IBuildContext> action, Bullseye.IBuildContext context) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput, Bullseye.IBuildContext> action, Bullseye.IBuildContext context) { }
    }
}