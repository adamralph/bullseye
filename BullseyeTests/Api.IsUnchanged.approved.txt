[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace Bullseye
{
    public class Options
    {
        public Options() { }
        public bool Clear { get; set; }
        public bool DryRun { get; set; }
        public bool ListDependencies { get; set; }
        public bool ListInputs { get; set; }
        public bool ListTargets { get; set; }
        public bool NoColor { get; set; }
        public bool ShowHelp { get; set; }
        public bool SkipDependencies { get; set; }
        public System.Collections.Generic.ICollection<string> Targets { get; }
        public bool Verbose { get; set; }
        public static string GetUsage(Bullseye.Internal.Palette p) { }
        public static Bullseye.Options Parse(System.Collections.Generic.IEnumerable<string> args) { }
    }
    public class static Targets
    {
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0. Will be removed in versio" +
            "n 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Func<System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Action action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static string[] DependsOn(params string[] dependencies) { }
        public static TInput[] ForEach<TInput>(params TInput[] inputs) { }
        [System.ObsoleteAttribute("Use RunTargets() instead. Will be removed in version 2.0.0.")]
        public static void Run() { }
        [System.ObsoleteAttribute("Use RunTargets() instead. Will be removed in version 2.0.0.")]
        public static void Run(System.Collections.Generic.IEnumerable<string> args) { }
        [System.ObsoleteAttribute("Use RunTargetsAsync() instead. Will be removed in version 2.0.0.")]
        public static System.Threading.Tasks.Task RunAsync(System.Collections.Generic.IEnumerable<string> args) { }
        [System.ObsoleteAttribute("Use RunTargetsAsync() instead. Will be removed in version 2.0.0.")]
        public static System.Threading.Tasks.Task RunAsync() { }
        public static void RunTargets() { }
        public static void RunTargets(System.Collections.Generic.IEnumerable<string> args) { }
        public static void RunTargets(Bullseye.Options options) { }
        public static System.Threading.Tasks.Task RunTargetsAsync(System.Collections.Generic.IEnumerable<string> args) { }
        public static System.Threading.Tasks.Task RunTargetsAsync(Bullseye.Options options) { }
        public static System.Threading.Tasks.Task RunTargetsAsync() { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target(string name) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        public static void Target(string name, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
    }
}